### PR TITLE
Update Godot to 4.2

### PR DESCRIPTION
**Pull request by Zishan-Rahman**

### CODE REVIEW

**Comment by Zishan-Rahman** [comment]
@Calinou I just tested it on my laptop and it seems to work as it did previously. Looks like this PR can be merged, and I'll merge [the relevant PR in GodotSharp](https://www.github.com/flathub/org.godotengine.GodotSharp#6) once this is merged as well :+1:

Some further clarification: I don't have commit and push rights to _this_ repo as of now so I can't merge this PR myself.

**Review by Calinou — APPROVED** [review]

**Pull request merged**
